pymc4/flow/executor.py

```diff
@@ -546,7 +552,12 @@ def proceed_distribution(
                 # might be posterior predictive or programmatically override to exchange observed variable to latent
                 if scoped_name not in state.untransformed_values:
                     # posterior predictive
-                    return_value = state.untransformed_values[scoped_name] = dist.sample()
+                    if dist.is_root:
```

Comment:
getting less and less difference from tfd coroutines

Reply:
Yeah, this PR brings us much closer to tfd coroutines. However, all of the "root" stuff and manual vectorization is optional. We do auto batching by default and the users don't have to think so much about all of the potential shape issues. So I think that we will be providing a much better UX than tfd coroutines

Reply:
+1, in tfd.jointCoroutine user need to identify the root, if we can do that automatically it is a win.

Reply:
Oh wait, we are doing that with `conditional_independent` kwarg... hmmm it is probably needed but could we find a better name for this kwarg?

Reply:
I think that calling them root is really vague. I think that they should be given a name that bears a meaning in probability distributions. I could only think of conditionally independent but maybe there's a more appropriate alternative